### PR TITLE
Adds gradle-dependency-graph-generator-plugin to the project

### PR DESCRIPTION
**Pull request by zugaldia**
This helps us visualize the dependency tree for Mapbox dependencies. To generate the chart, run the following Gradle task:

```
$ ./gradlew generateDependencyGraphMapboxLibraries
```

Graphviz needs to be installed following the instructions on the [plugin README](https://github.com/vanniktech/gradle-dependency-graph-generator-plugin).

cc: @langsmith @Guardiola31337

### CODE REVIEW

File: MapboxAndroidDemo/build.gradle [hunk 1]
```diff
@@ -2,6 +2,7 @@ apply plugin: 'com.android.application'
 apply plugin: 'com.google.firebase.firebase-perf'
 apply plugin: 'com.github.triplet.play'
 apply from: "$project.rootDir/gradle/script-git-version.gradle"
+apply from: "$project.rootDir/gradle/graph.gradle"
```
Comment (Guardiola31337) [inline MapboxAndroidDemo/build.gradle]: In order to understand in a glance what the plugin does, what about calling it `dependencies-graph.gradle` instead?

File: gradle/graph.gradle [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.vanniktech.dependency.graph.generator.dot.Color
+import com.vanniktech.dependency.graph.generator.dot.Shape
+import com.vanniktech.dependency.graph.generator.dot.Style
+
+plugins.apply(DependencyGraphGeneratorPlugin)
+
+def mapboxGenerator = new Generator(
+        "mapboxLibraries", // Suffix for our Gradle task.
+        "", // Root suffix that we don't want in this case.
+        { dependency -> dependency.getModuleGroup().startsWith("com.mapbox") }, // Only want Mapbox libs.
```
Comment (Guardiola31337) [inline gradle/graph.gradle]: Our `GROUP` is `com.mapbox.mapboxsdk`. Shouldn't we use the complete group here?

File: gradle/graph.gradle [hunk 3]
```diff
@@ -0,0 +1,29 @@
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath "com.vanniktech:gradle-dependency-graph-generator-plugin:0.3.0"
```
Comment (Guardiola31337) [inline gradle/graph.gradle]: For consistency we should add this dependency in the [dependencies.gradle](https://github.com/mapbox/mapbox-android-demo/blob/master/gradle/dependencies.gradle) file.

Comment (zugaldia) [reply gradle/graph.gradle]: This is the same approach `script-git-version.gradle` follows. If we're gonna change one script for consistency, we should change them all - feel free to cut a separate ticket.

File: gradle/graph.gradle (hunk 2 above)
Comment (zugaldia) [reply gradle/graph.gradle]: Changed, although it doesn't have an effect in the check.

File: MapboxAndroidDemo/build.gradle (hunk 1 above)
Comment (zugaldia) [reply MapboxAndroidDemo/build.gradle]: Renamed to `gradle-dependencies-graph.gradle`. Looks like the Maps SDK likes prefixing all scripts with `gradle-`.

**Comment by zugaldia** [comment]
@Guardiola31337 @tobrun Ready for another round of 👀 .

File: gradle/graph.gradle (hunk 3 above)
Comment (Guardiola31337) [reply gradle/graph.gradle]: Follow up ticket 👉 https://github.com/mapbox/mapbox-android-demo/issues/676

**Pull request merged**